Lista_func.c

```diff
@@ -0,0 +1,112 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include "Lista_funcc.h"
+
+list* init() //Funkcja inicjalizacji listy
+{
+	list* HEAD = (list*)malloc(sizeof(list));
+	HEAD->head = NULL;
+	return HEAD;
+}
+
+int insert(list* header, int i) //Funkcja dodajaca jeden NODE zawierajacego liczbe naturalna oraz adres nastepnej komorki pamieci
+{
+	if (header->head == NULL)
+	{
+		header->head = (node*)malloc(sizeof(node));
```

Comment:
Nie trzeba rzutowac na node'a
```suggestion
		header->head = malloc(sizeof(node));
```